File: PhotonCore/src/main/java/com/outoftheboxrobotics/photoncore/PhotonCommandBase.java
package com.outoftheboxrobotics.photoncore;

import com.qualcomm.hardware.lynx.LynxNackException;
import com.qualcomm.hardware.lynx.commands.LynxMessage;

import java.util.concurrent.CompletableFuture;

public interface PhotonCommandBase<RESPONSE extends LynxMessage>{
    CompletableFuture<LynxMessage> getResponse();
    void send() throws InterruptedException, LynxNackException;
    RESPONSE sendReceive() throws InterruptedException, LynxNackException;
}

File: PhotonCore/src/main/java/com/outoftheboxrobotics/photoncore/HAL/PhotonCommandBase.kt
package com.outoftheboxrobotics.photoncore.HAL

import com.qualcomm.hardware.lynx.LynxNackException
import com.qualcomm.hardware.lynx.commands.LynxMessage
import java.util.concurrent.CompletableFuture

interface PhotonCommandBase {
    fun getResponse(): CompletableFuture<LynxMessage>
}